operator/v1/types_ingress.go

```diff
@@ -545,6 +545,20 @@ type AWSLoadBalancerParameters struct {
 	//
 	// +optional
 	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+
+	// subnets specifies the list of subnets for the load balancer to
+	// route traffic to. The values can be either a subnet ID or name.
```

Comment:
how can we validate these?

Reply:
> how can we validate these?

One common option for "either/or" choices is a discriminated union, so instead of string this would be a struct. And the it's a question of whether the user can mix and match subnetIDs and names or if they're all one type or the other.

Reply:
We can provide both at the same time. The annotation `service.beta.kubernetes.io/aws-load-balancer-subnets` accepts both, within the same annotation (side by side), and here's the function that determines whether it's a ID or name: https://github.com/openshift/cloud-provider-aws/blob/b65e315db19205593d399af6e636d71ae2b1eb1c/pkg/providers/v1/aws.go#L3723

So, given the annotations existing flexibility, is there much value in validating the values or applying more structure?

Reply:
BTW the API has been updated to add validation for max subnet id/name length, determine by the subnet name, which is an AWS tag, that cannot be longer than 256 chars.

Reply:
Can you find anything that suggests only specific characters are allowed in either the name or ID?

Reply:
I responded to this in the EP actually https://github.com/openshift/enhancements/pull/1595#discussion_r1573440161:

> A set of characters?

I do see that https://docs.aws.amazon.com/tag-editor/latest/userguide/tagging.html says:

> In general, the allowed characters are letters, numbers, spaces representable in UTF-8, and the following characters: _ . : / = + - @.

However, in testing, I can't find an invalid character for subnet tag names. gspence-test-!@#$%^&*()``~<>?:"{}|_+ was valid for a subnet name. I don't think we can limit the set of characters.

Reply:
FWIW I found a reference on tags in https://docs.aws.amazon.com/mediaconnect/latest/ug/tagging-restrictions.html, but I couldn't find anything similar for VPC.

Reply:
In https://docs.aws.amazon.com/tag-editor/latest/userguide/tagging.html it also says that `tag policies` can be created by organizations, so we likely cannot exert any control over this that would conflict with a customer `tag policy`.